pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -371,6 +377,18 @@ func (b *serviceBinder) setApplicationObjects(
 	sbrStatus.Applications = boundApps
 }
 
+// set default value for application selector
+func ensureDefaults(applicationSelector *v1alpha1.ApplicationSelector) {
+	if applicationSelector.LabelSelector == nil {
+		applicationSelector.LabelSelector = &metav1.LabelSelector{}
+	}
+	if applicationSelector.BindingPath == nil {
```

Comment:
Is it valid if `applicationSelector.BindingPath` is not `nil` but `ContainersPath` is `""` (empty string)?

Reply:
Yes.

There is a check before calling update:
https://github.com/redhat-developer/service-binding-operator/blob/be37227d54ed101c4e57a427fea93abc1cf54cdb/pkg/controller/servicebindingrequest/binder.go#L524

Reply:
I wonder whether this will affect the current default behavior where the user doesn't have to specify `bindingPath.containersPath`; at the moment they try to use the secret injection feature they'll need to additionally specify `bindingPath.containersPath` since if `bindingPath` is not `nil` and `containersPath` is empty, effectively disabling the existing containers path injection. Or perhaps I'm missing something?

Reply:
In the current code, irrespective of `containerPath` is specified or not, based on this check `secretPath` is going to get updated.